lib/std/fs/test.zig

```diff
@@ -1418,3 +1415,22 @@ test "File.PermissionsUnix" {
     try testing.expect(permissions_unix.unixHas(.user, .execute));
     try testing.expect(!permissions_unix.unixHas(.other, .execute));
 }
+
+test "delete a read-only file on windows" {
+    if (builtin.os.tag != .windows) return error.SkipZigTest;
+
+    var tmp = tmpDir(.{});
+    defer tmp.cleanup();
+    const file = try tmp.dir.createFile("test_file", .{ .read = true });
+    // Create a file and make it read-only
+    const metadata = try file.metadata();
+    var permissions = metadata.permissions();
+    permissions.setReadOnly(true);
+    try file.setPermissions(permissions);
+    try testing.expectError(error.AccessDenied, tmp.dir.deleteFile("test_file"));
```

Comment:
It doesn't produce `AccessDenied` in this test case. I think it puts it in Pending state, until the close (open count goes back to 0 *).  

However, not closing the handle at all will produce an unexpected error with `DELETE_PENDING`. Not sure how we should handle that.

Reply:
I assume that comes during `tmp.cleanup` when it tries to delete the tmp tree? Perhaps `DeleteFile` should treat `DELETE_PENDING` as if the delete succeeded (i.e. `.DELETE_PENDING => return,` in the `NtCreateFile` rc switch)?

Reply:
Correct,
that seems like the only solution tbh since I don't think `DELETE_PENDING` always means unhandled close. 
 At least we would know that the directory is not empty once it tries to remove the tmp dir.

I will do some testing on that and probably commit afterwards.